src/dec128.rs

```diff
@@ -104,16 +104,15 @@ struct d128Visitor;
 impl serde::de::Visitor for d128Visitor {
     type Value = d128;
 
-    fn visit_str<E>(&mut self, s: &str) -> Result<d128, E>
+    fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
```

Comment:
It will print a message like

```
Syntax(Message("invalid type: map, expected a d128 value"), 1, 25)
```

Already fixed in the latest commit. Sorry :(